storage/suffix_test.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"bytes"
+	"encoding/base64"
+	"math/big"
+	"testing"
+)
+
+const (
+	logStrataDepth = 8
+	maxLogDepth    = 64
+)
+
+func TestSuffixKeyEquals(t *testing.T) {
+	for _, tc := range []struct {
+		prefix    []byte
+		leafIndex int64
+		want      []byte
+	}{
+		{h2b(""), 1, h2b("0801")},
+		{h2b("00"), 1, h2b("0801")},
+	} {
+		sfxA, err := makeSuffixKey(logStrataDepth, tc.leafIndex)
+		if err != nil {
+			t.Errorf("makeSuffixKey(%v, %v): %v", logStrataDepth, tc.leafIndex, err)
+			continue
+		}
+
+		sfxABytes, err := base64.StdEncoding.DecodeString(sfxA)
+		if err != nil {
+			t.Errorf("makeSuffixKey(%v, %v): %v", logStrataDepth, tc.leafIndex, err)
+			continue
+		}
+
+		if got, want := sfxABytes, tc.want; !bytes.Equal(got, want) {
+			t.Errorf("makeSuffixKey(%v, %v): %x, want %x", logStrataDepth, tc.leafIndex, got, want)
+			continue
+		}
+
+		nodeID := NewNodeIDFromPrefix(tc.prefix, logStrataDepth, tc.leafIndex, logStrataDepth, maxLogDepth)
+		_, sfxB := nodeID.Split(len(tc.prefix), logStrataDepth)
+		sfxBKey := sfxB.Serialize()
+		sfxBBytes, err := base64.StdEncoding.DecodeString(sfxBKey)
+		if err != nil {
+			t.Errorf("splitNodeID(%v): _, %v", nodeID, err)
+			continue
+		}
+
+		if got, want := sfxBBytes, tc.want; !bytes.Equal(got, want) {
+			t.Errorf("[%x, %v].splitNodeID(%v, %v): %v.Serialize(): %x, want %x", nodeID.Path, nodeID.PrefixLenBits, len(tc.prefix), logStrataDepth, sfxB, got, want)
+			continue
+		}
+	}
+}
+
+func TestSuffixKey(t *testing.T) {
+	for _, tc := range []struct {
+		depth   int
+		index   int64
+		want    []byte
+		wantErr bool
+	}{
+		{depth: 0, index: 0x00, want: h2b("0000"), wantErr: false},
+		{depth: 8, index: 0x00, want: h2b("0800"), wantErr: false},
+		// TODO(gdbelvin): want: "0fabcd"?
```

Comment:
Kindof...  

`makeSuffixKey` has been deprecated in this PR in favor of `NodeID.Split()`.  To prove to myself that I was keeping the exact same behavior, however, I moved `makeSuffixKey` to `suffix_test.go` and added this test to document its behavior. Because `makeSuffixKey` assumed a maximum subtree depth of 8, it caused problems for the map, which has deep subtrees. 

All these issues have been resolved with the migration to `NodeID.Split()`, which produces the correct output for both log and map sub protos -- and maintains the exact same byte format for the Log's cases.